### PR TITLE
add the optional fallback consumer

### PR DESCRIPTION
**Pull request by elakito**
PR for https://github.com/go-openapi/runtime/issues/104
thanks.
regards, aki

### CODE REVIEW

File: client/runtime_test.go [hunk 1]
```diff
@@ -808,3 +808,46 @@ func TestRuntime_PreserveTrailingSlash(t *testing.T) {
 
 	assert.NoError(t, err)
 }
+
+func TestRuntime_FallbackConsumer(t *testing.T) {
+	result := `W3siY29tcGxldGVkIjpmYWxzZSwiY29udGVudCI6ImRHRnpheUF4SUdOdmJuUmxiblE9IiwiaWQiOjF9XQ==`
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.Header().Add(runtime.HeaderContentType, "application/x-task")
+		rw.WriteHeader(http.StatusOK)
+		rw.Write([]byte(result))
```
Comment (golangci) [inline client/runtime_test.go]: Error return value of `rw.Write` is not checked

**Comment by elakito** [comment]
i'll resolve the issue...

File: client/runtime_test.go [hunk 2]
```diff
@@ -808,3 +808,71 @@ func TestRuntime_PreserveTrailingSlash(t *testing.T) {
...
+
+	rwrtr := runtime.ClientRequestWriterFunc(func(req runtime.ClientRequest, _ strfmt.Registry) error {
+		return req.SetBodyParam(bytes.NewBufferString("hello"))
+	})
+
+	hu, _ := url.Parse(server.URL)
+	rt := New(hu.Host, "/", []string{"http"})
+
+	// without the fallback consumer
+	res, err := rt.Submit(&runtime.ClientOperation{
```
Comment (golangci) [inline client/runtime_test.go]: ineffectual assignment to `res`

**Comment by elakito** [comment]
@casualjim ahh. I didn't see your reply about how to pass the check. I made some change and I hope it would pass the check this time.

**Comment by elakito** [comment]
oh. no. golangci is very picky...

**Pull request merged**